trl/trainer/accelerate_ppo.py

```diff
@@ -0,0 +1,423 @@
+                      clip_by_value,
+                      entropy_from_logits,
+                      flatten_dict,
+                      stats_to_np,
+                      stack_dicts,
+                      WANDB_PADDING)
+from trl.trainer import BaseTrainer, AdaptiveKLController, FixedKLController, LengthSampler
+from trl import AutoModelForCausalLMWithValueHead
+
+
+class AcceleratePPOTrainer(BaseTrainer):
+    """
+    The PPO_trainer uses Proximal Policy Optimization to optimise language models.
+    """
+
+    default_params = {
+        "lr": 1.41e-5,
+        "adap_kl_ctrl": True,
+        "init_kl_coef":0.2,
+        "target": 6,
+        "horizon":10000,
+        "gamma":1,
+        "lam":0.95,
+        "cliprange": .2,
+        "cliprange_value":.2,
+        "vf_coef":.1,
+        "batch_size": 256,
+        "forward_batch_size": 16,
+        "ppo_epochs": 4,
+    }
+
+    def __init__(self, model=None, ref_model=None, tokenizer=None, **config):
+        """
+        Initialize PPOTrainer.
+        Args:
+            model (torch.model): Hugging Face transformer GPT2 model with value head
+            ref_model (torch.model): Hugging Face transformer GPT2 refrence model used for KL penalty
+            tokenizer (tokenizer): Hugging Face tokenizer
+            ppo_params (dict or None): PPO parameters for training. Can include following keys:
+                'lr' (float): Adam learning rate, default: 1.41e-5
+                'batch_size' (int): Number of samples per optimisation step, default: 256
+                'forward_batch_size' (int): Number of samples forward passed through model at a time, default: 16
+                'ppo_epochs' (int): Number of optimisation epochs per batch of samples, default: 4
+                'gamma' (float)): Gamma parameter for advantage calculation, default: 1.
+                'lam' (float): Lambda parameter for advantage calcualation, default: 0.95
+                'cliprange_value' (float): Range for clipping values in loss calculation, default: 0.2
+                'cliprange' (float): Range for clipping in PPO policy gradient loss, default: 0.2
+                'vf_coef' (float): Scaling factor for value loss, default: 0.1
+                'adap_kl_ctrl' (bool): Use adaptive KL control, otherwise linear, default: True
+                'init_kl_coef' (float): Initial KL penalty coefficient (used for adaptive and linear control), default: 0.2
+                'target' (float): Target KL value for adaptive KL control, default: 6.0
+                'horizon' (float): Horizon for adaptive KL control, default: 10000
+        """
+        super().__init__(self.default_params)
+
+        # Step 1: Initialize Accelerator
+        self.accelerator = Accelerator(log_with="wandb")
+        self.config.update(config)
+
+        # Step 2: Initialize model, tokenizer and dataset        
+        self._build_models_and_tokenizer(model, ref_model, tokenizer)
+        self._build_dataset()
```

Comment:
this is not needed, it's the user's responsibility to build their own dataset

Reply:
Removed in [7516b37](https://github.com/lvwerra/trl/pull/58/commits/7516b37181ded8311bfaad33802829e9767e3487)